extensions/ql-vscode/src/view/result-tables.tsx

```diff
@@ -70,13 +68,13 @@ function renderResultCountString(resultSet: ResultSet): JSX.Element {
 export class ResultTables
   extends React.Component<ResultTablesProps, ResultTablesState> {
 
-  private getResultSets(): ResultSet[] {
+  private static _getResultSetsOfProps(props: ResultTablesProps): ResultSet[] {
```

Comment:
Why convert to static?

Reply:
So that it can be called from the static lifecycle method `getDerivedStateFromProps`.